### ISSUE TITLE
[BUG] OS Updating Breaks HashCheck

### ISSUE THREAD

**Issue by CanadianJeff** [gurnec/HashCheck#15]
my windows 10 just updated to the latest version and now I have a "Windows.old"

hashcheck was left behind in "Windows.old" I found this out when I started autoruns from sysinternals autoruns and it showed that the dll files for hashcheck were missing

installing a new copy of hashcheck put the new files in place so it works again just stupid microsoft being dumb thats all

**Comment by gurnec** [comment]
To be fair, HashCheck probably shouldn't be installed under \Windows... but when it was written back in 2008, it wasn't uncommon practice to place files there, and it's never occurred to me that this should change.

I'll leave this open, it should probably be fixed. Thanks for catching it, and for checking in autoruns that it was still in the registry after the upgrade.

**Comment by CanadianJeff** [comment]
screenshot of my current autoruns here
https://puu.sh/rjwVt.png

**Comment by Masxx** [comment]
Can confirm, happened to me also when I received the Anniversary update for Windows 10.

**Comment by KrasnayaPloshchad** [comment]
To avoid OS update breaking this extension, the installation path should be `C:\Program Files\HashCheck`.

**Comment by CanadianJeff** [comment]
ah god damn it windows I just updated to the new creators update and now my hashcheck is gone once again thank you microsoft :-)

**Comment by scara** [comment]
Me too.

**Comment by ghost** [comment]
So, uh... Is this getting changed? I love this extension :D

**Comment by R-Adrian** [comment]
upgraded to Windows 10 1709 (release preview) - HashCheck is trashed yet again and i had to reinstall it.

please upvote this feedback for MS:  https://aka.ms/I6yxql  @

**Comment by JohnLGalt** [comment]
Hi,

As a Windows Insider Preview tester, this happens to me quite a lot.  

While it is simple to just reinstall, I'm still adding my comment (more so to to _nudge_ you just a bit on this issue than anything else).

**Comment by R-Adrian** [comment]
Windows 10 1803 - Spring Creators Update (build 17133.1)  breaks HashCheck v2.4.0 again, predictably.

had to reinstall HashCheck after the edition upgrade.

**Comment by ZPNRG** [comment]
Yes, as Aditza2015 keeps bringing up, every Windows 10 major update breaks HashCheck v2.4.0 and you have to reinstall it.  It is easy and fast to reinstall, but would be nice to see this fixed.

**Comment by JohnLGalt** [comment]
I've gotten used to this, as I run Insider Preview builds (and Skip ahead builds when they are available) on my testing desktop, and every build change causes HC to break.  But it would be nice if this could survive OS upgrades.

**Comment by jo2k** [comment]
I also had to reinstall after upgrading to Windows 10 1803. Program shows in app/remove programs, but attempting to uninstall it throws an error.
